doc/usage.adoc

```diff
@@ -232,177 +116,35 @@ Ranges of timestamps    Message
-https://en.wikipedia.org/wiki/List_of_ISO_3166_country_codes[ISO country
-code] format an error will be thrown making
-`+SensorView.global_ground_truth+` invalid because
-`+SensorView.global_ground_truth.country_code+` is not set. The
-incorrectness is appearing in the intervals between message frame 0 and
-message frame 498 but not in message frame 499. In the message frame 499
-the `+GroundTruth.country_code+` is set but do not comply to the
-`+is_iso_country_code+` rule. That is why you see split frame messages
-like this [0, 498], [500, 998] for not set and 499 for is not ISO
-country code. Note that `+GroundTruth.country_code+` refers to the same
-path as `+SensorView.global_ground_truth.country_code+`. The SensorView
-part is cut due to better readability.
-
-In the output there are more message fields which are not set on the
-`+GroundTruth+` level making it invalid:
-
-[source,bash]
-----
-[0, 2717]                                 GroundTruth.proj_string.is_set(None) does not comply in SensorView.global_ground_truth
-[0, 2717]                                 GroundTruth.map_reference.is_set(None) does not comply in SensorView.global_ground_truth
-[0, 2717]                                 GroundTruth.occupant.is_set(None) does not comply in SensorView.global_ground_truth
-[0, 2717]                                 GroundTruth.traffic_sign.is_set(None) does not comply in SensorView.global_ground_truth
-[0, 2717]                                 GroundTruth.traffic_light.is_set(None) does not comply in SensorView.global_ground_truth
-[0, 2717]                                 GroundTruth.road_marking.is_set(None) does not comply in SensorView.global_ground_truth
-----
-
-Next the path `+GroundTruth.environmental_conditions+` is set but not
-valid leading to the output below (Note that the indentation
-demonstrates the hierarchy of the message fields):
-
-[source,bash]
-----
-[0, 2717]                                 GroundTruth.environmental_conditions.is_valid(None) does not comply in SensorView.
-    [0, 2717]                                 EnvironmentalConditions.atmospheric_pressure.is_greater_than_or_equal_to(80000) does not comply in SensorView.global_ground_truth.environmental_conditions.atmospheric_pressure
-    [0, 2717]                                 EnvironmentalConditions.temperature.is_greater_than_or_equal_to(170) does not comply in SensorView.global_ground_truth.environmental_conditions.temperature
-    [0, 2717]                                 EnvironmentalConditions.unix_timestamp.is_set(None) does not comply in SensorView.global_ground_truth.environmental_conditions
-    [0, 498], [500, 998], [1000, 1498],       EnvironmentalConditions.fog.is_set(None) does not comply in SensorView.global_ground_truth.environmental_conditions
-    [1500, 1998], [2000, 2498], [2500, 2716]
-----
-
-The output is generate because of the rules defined in
-`+osi_environment.yml+`:
-
-[source,yaml]
-----
-EnvironmentalConditions:
```

Comment:
Don't you think that we have to explain how to work with these rules?

Reply:
The explanation of how to write rules is in "writing-rules.adoc". Here I added some explanation of the rules that are given as errors in the example trace file and removed the rest.

For the future, I would create a new trace file containing more examples of rule violations and then explain them here. But I would do that after the planned major rework of osi-validation.